hs_geographic_feature_resource/tests/test_geo_feature.py

```diff
@@ -0,0 +1,580 @@
+__author__ = 'drew'
+
+from unittest import TestCase
+from django.contrib.contenttypes.models import ContentType
+from hs_core.hydroshare import utils, users, resource
+from hs_core.models import GenericResource, Creator, Contributor, CoreMetaData, \
+    Coverage, Rights, Title, Language, Publisher, Identifier, \
+    Type, Subject, Description, Date, Format, Relation, Source
+
+from hs_geographic_feature_resource.models import *
+from hs_geographic_feature_resource.receivers import *
+from django.contrib.auth.models import Group, User
+from hs_core import hydroshare
+from django.core.files.uploadedfile import UploadedFile
+from django.http import HttpRequest, QueryDict
+
+class TestGeoFeatureMetadata(TestCase):
+    def setUp(self):
+        try:
+            #self.user = User.objects.create_user('user1', email='user1@nowhere.com')
+            self.group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+            self.user = hydroshare.create_account(
+                'zhiyu.li@byu.edu',
+                username='drew',
+                first_name='Zhiyu',
+                last_name='Li',
+                superuser=False,
+                groups=[self.group]
+            )
+            self.allowance=0.00001
+        except:
+            self.tearDown()
+            self.user = User.objects.create_user('user1', email='user1@nowhere.com')
+
+
+        self.resGeoFeature = hydroshare.create_resource(
+            resource_type='GeographicFeatureResource',
+            owner=self.user,
+            title='Test Geographic Feature (shapefiles)',
+            keywords=['kw1', 'kw2']
+        )
+    def tearDown(self):
+        User.objects.all().delete()
+        Group.objects.all().delete()
+        GenericResource.objects.all().delete()
+        Creator.objects.all().delete()
+        Contributor.objects.all().delete()
+        CoreMetaData.objects.all().delete()
+        Coverage.objects.all().delete()
+        Rights.objects.all().delete()
+        Title.objects.all().delete()
+        Publisher.objects.all().delete()
+        Description.objects.all().delete()
+        Relation.objects.all().delete()
+        Subject.objects.all().delete()
+        Source.objects.all().delete()
+        Identifier.objects.all().delete()
+        Type.objects.all().delete()
+        Format.objects.all().delete()
+        Date.objects.all().delete()
+        Language.objects.all().delete()
+        #remove res specific model objs here
+        #Variable.objects.all().delete()
+        OriginalCoverage.objects.all().delete()
+        GeometryInformation.objects.all().delete()
+        FieldInformation.objects.all().delete()
+        OriginalFileInfo.objects.all().delete()
```

Comment:
Django's own `TestCase` already rolls the database back after each test, so the `tearDown()` doesn't need to do all this